src/transform.ts

```diff
@@ -177,12 +168,54 @@ function concatArrays(...arrs: ArrayLike<number>[]) {
   return c
 }
 
-function extractAndNormaliseTransaction(
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const objectToMap = (object: Record<string, any>): Map<string, unknown> => {
+  return new Map(
+    Object.entries(object).map(([key, value]) => {
+      if (key === 'r' && value instanceof Map && Array.from(value.keys()).every((k) => typeof k === 'number')) {
+        return [key, value]
+      }
+      if (value instanceof Uint8Array) {
+        if (['snd', 'close', 'aclose', 'rekey', 'rcv', 'arcv', 'fadd', 'asnd', 'm', 'r', 'f', 'c'].includes(key) && value.length === 32) {
```

Comment:
I'm a little worried about maintenance of this code - what if new fields are added to the protocol in the future and we miss them? I'm a bit unclear on how we tackled this previously from the diff.

Reply:
There was a change to `algosdk` which made address handling a bit tricky for us. For context, we receive address data as Uint8Array (length 32) from algod API. In the past, `algosdk` allowed the address fields to be Uint8Array but now it enforce all addresses to be type Address or string. Additionally, `algosdk` `Transaction.fromEncodingData` expects us to past in a Map object. That was why I had to convert the `BlockTransaction` in to a Map & hard-coded all the address field names.
I know this isn't the cleanest way to do ... I hope the above explanation gives some context, I'm open for suggestions here.